crypto/bio/b_sock2.c

```diff
@@ -115,6 +115,57 @@ int BIO_connect(int sock, const BIO_ADDR *addr, int options)
     return 1;
 }
 
+/*-
+ * BIO_localaddr - Sets local address for socket
+ * @sock: the socket to set
+ * @addr: local address to bind to
+ * @options: BIO socket options
+ *
+ * Binds to the address using the given socket and options, then
+ * starts listening for incoming connections.
+ *
+ * Options can be a combination of the following:
+ * - BIO_SOCK_REUSEADDR: Try to reuse the address and port combination
+ *   for a recently closed port.
+ *
+ * When restarting the program it could be that the port is still in use.  If
+ * you set to BIO_SOCK_REUSEADDR option it will try to reuse the port anyway.
+ * It's recommended that you use this.
+ */
+int BIO_localaddr(int sock, const BIO_ADDR *addr, int options)
+{
+    int on = 1;
+    int socktype;
+
+    if (sock == -1) {
+        BIOerr(BIO_F_BIO_LISTEN, BIO_R_INVALID_SOCKET);
```

Comment:
BIO_F_BIO_LISTEN is supposed to mean that error occurred in BIO_listen subroutine.

Reply:
So I should add a BIO_F_BIO_LOCALADDR?

Reply:
Yes - add it here and then run "make update", which will set up all the necessary defines.

Reply:
Do "make errors" which will point out that the define is wrong and then "make update" will create the define, and then you can use it.